shared/eventbus/eventbus.go

```diff
@@ -75,32 +94,48 @@ func (b *BrokerService) Register(topic, description string, typ any) error {
 }
 
 func (b *BrokerService) EventTypes() []EventType {
+	// there's really no race condition here, but we need this lock to pass race condition tests.
+	b.eventMutex.Lock()
 	types := make([]EventType, 0, len(b.eventTypes))
 	for _, event := range b.eventTypes {
 		types = append(types, event)
 	}
+	b.eventMutex.Unlock()
+
 	sort.Slice(types, func(i, j int) bool {
 		return types[i].Topic < types[j].Topic
 	})
 	return types
 }
 
 func (b *BrokerService) Publish(topic string, ctx request.CTX, data any) error {
-	// there's really no race condition here so no mutex required
+	// no need for mutex since no race condition here
+	// this is done not to receive events after Stop() was called
+	// Since here we have not global mutex(for performance reasons) "couple of"
+	// events might pass through even after calling Stop(), but it should be Ok.
+	stopping := atomic.LoadInt32(&b.stopping) == STOPPING
+	if stopping {
+		return errors.New("event bus was stopped")
+	}
+	// there's really no race condition here, but we need this lock to pass race condition tests.
+	b.eventMutex.Lock()
```

Comment:
We should put this whole check into it's own method.

Reply:
Good point!